desktop/apps/article2/components/EditButton.js

```diff
@@ -0,0 +1,52 @@
+import PropTypes from 'prop-types'
+import { pluck, contains } from 'underscore'
+import * as React from 'react'
+import { data as sd } from 'sharify'
+import ChannelMemberQuery from 'desktop/apps/article2/queries/channelMember'
+import positronql from 'desktop/lib/positronql.coffee'
+import styled from 'styled-components'
+
+export default class EditButton extends React.Component {
+  static propTypes = {
+    channelId: PropTypes.string,
+    slug: PropTypes.string
+  }
+
+  constructor (props) {
+    super(props)
+    this.state = {
+      hasButtonState: false
+    }
+  }
+
+  componentDidMount = async () => {
+    const userId = sd.CURRENT_USER && sd.CURRENT_USER.id
+    if (!userId) {
+      return this.setState({ showEditButton: false, hasButtonState: true })
+    }
+    const data = await positronql({ query: ChannelMemberQuery(userId) })
```

Comment:
Niptick—it can help readability to pull out more specific variable names e.g.

```javascript
const { channels } = await positronql(...)
const hasChannelAccess = contains(pluck(...))
if (hasChannelAccess) {
  // ...
}
```